WcaOnRails/app/inputs/money_amount_input.rb

```diff
@@ -3,14 +3,15 @@ class MoneyAmountInput < SimpleForm::Inputs::Base
   def input(wrapper_options)
     merged_input_options = merge_wrapper_options(input_html_options, wrapper_options)
 
-    actual_field = @builder.hidden_field(attribute_name)
+    value = options.delete(:value) || @builder.object.send(attribute_name)
+    actual_field = @builder.hidden_field(attribute_name, value: value)
     input_id = attribute_name.to_s + "_input_field"
 
     # On page load, inputs with this class get their mask setup
     merged_input_options[:class] << "wca-currency-mask"
 
     amount_input = template.content_tag(:input, "",
-                                        value: @builder.object.send(attribute_name),
+                                        value: value,
```

Comment:
haha, and you didn't have to write the class :p
I actually had to dig into simpleform's code to know how some things worked...
I added a bunch of comments, hopefully it will clarify the code for the reader.